src/Params/ReportMakeParams.php

```diff
@@ -0,0 +1,233 @@
+     *
+     * @var string
+     */
+    private $type;
+
+    /**
+     * Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ)).
+     *
+     * @var string
+     */
+    private $value;
+
+    /**
+     * Additional request options.
+     *
+     * @var array<mixed>|null
+     */
+    private $options = [];
+
+    /**
+     * Force update report, if it already was generated previously.
+     *
+     * @var bool
+     */
+    private $is_force = false;
+
+    /**
+     * URL to call (using `post` method) when report content updated.
+     *
+     * @var string|null
+     */
+    private $on_update_url;
+
+    /**
+     * URL to call (using `post` method) when report generation completed.
+     *
+     * @var string|null
+     */
+    private $on_complete_url;
+
+    /**
+     * Additional request data.
+     *
+     * @var array<mixed>|null
+     */
+    private $data;
+
+    /**
+     * Idempotence key which the server uses to recognize subsequent retries of the same request.
+     *
+     * @var string|null
+     */
+    private $idempotence_key;
+
+    /**
+     * ReportMakeParameter constructor.
+     *
+     * @param string $report_type_uid Unique report type ID (e.g.: `some_report_uid` or `some_report_uid@domain`)
+     * @param string $type            Request type (e.g.: `VIN`, `GRZ`, `STS`, `PTS`, `CHASSIS`, etc.)
+     * @param string $value           Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ))
+     */
+    public function __construct(string $report_type_uid, string $type, string $value)
+    {
+        $this->report_type_uid = $report_type_uid;
+        $this->type            = $type;
+        $this->value           = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getReportTypeUid(): string
+    {
+        return $this->report_type_uid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param array<mixed>|null $options
+     *
+     * @return $this
+     */
+    public function setOptions(?array $options): self
+    {
+        $this->options = $options;
+
+        return $this;
+    }
+
+    /**
+     * @return array<mixed>|null
+     */
+    public function getOptions(): ?array
+    {
+        return $this->options;
+    }
+
+    /**
+     * @param bool $is_force
+     *
+     * @return $this
+     */
+    public function setForce(bool $is_force): self
+    {
+        $this->is_force = $is_force;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isForce(): bool
+    {
+        return $this->is_force;
+    }
+
+    /**
+     * @param string|null $on_update_url
+     *
+     * @return $this
+     */
+    public function setOnUpdateUrl(?string $on_update_url): self
```

Comment:
why `$on_update_url` is optional?

Reply:
For all previously available properties, I relied on the signature of the old method.

Earlier:

```php
?string $on_update = null
```

Reply:
Removed the optionality of the parameter.